generatebundlefile/helm.go

```diff
@@ -0,0 +1,147 @@
+func helmLog(log logr.Logger) action.DebugLog {
+	return func(template string, args ...interface{}) {
+		log.Info(fmt.Sprintf(template, args...))
+	}
+}
+
+// UnTarHelmChart will attempt to move the helm chart out of the helm cache, by untaring it to the pwd and creating the filesystem to unpack it into.
+func UnTarHelmChart(chartRef, chartPath, dest string) error {
+	if chartRef == "" || chartPath == "" || dest == "" {
+		return fmt.Errorf("Empty input value given for UnTarHelmChart")
+	}
+	_, err := os.Stat(dest)
+	if os.IsNotExist(err) {
+		if _, err := os.Stat(chartPath); err != nil {
+			if err := os.MkdirAll(chartPath, 0755); err != nil {
+				return errors.Wrap(err, "failed to untar (mkdir)")
+			}
+		} else {
+			return errors.Errorf("failed to untar: a file or directory with the name %s already exists", dest)
+		}
+	}
+	// Untar the files, and create the directory structure
+	return chartutil.ExpandFile(dest, chartRef)
+}
+
+// hasRequires checks for the existance of the requires.yaml within the helm directory
+func hasRequires(helmdir string) (string, error) {
+	requires := filepath.Join(helmdir, "requires.yaml")
+	info, err := os.Stat(requires)
+	if os.IsNotExist(err) {
+		return "", err
+	}
+	if info.IsDir() {
+		return "", fmt.Errorf("Found Dir, not requires.yaml file")
+	}
+	return requires, nil
+}
+
+// validateHelmRequires runs the parse file into struct function, and validations
+func validateHelmRequires(fileName string) (*Requires, error) {
+	helmrequires := &Requires{}
+	err := parseHelmRequires(fileName, helmrequires)
+	if err != nil {
+		return nil, err
+	}
+	err = validateHelmRequiresContent(helmrequires)
+	if err != nil {
+		return nil, err
+	}
+	return helmrequires, err
+}
+
+// validateHelmRequiresContent loops over the validation tests
+func validateHelmRequiresContent(helmrequires *Requires) error {
+	for _, v := range helmRequiresValidations {
+		if err := v(helmrequires); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+var helmRequiresValidations = []func(*Requires) error{
+	validateHelmRequiresName,
+}
+
+func validateHelmRequiresName(helmrequires *Requires) error {
+	err := helmrequires.validateHelmRequiresNotEmpty()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// validateHelmRequiresNotEmpty checks that it has at least one image in the spec
+func (helmrequires *Requires) validateHelmRequiresNotEmpty() error {
+	// Check if Projects are listed
+	if len(helmrequires.Spec.Images) < 1 {
+		return fmt.Errorf("Should use non-empty list of images for requires")
+	}
+	return nil
+}
+
+// parseHelmRequires will attempt to unpack the requires.yaml into the Go struct `Requires`
+func parseHelmRequires(fileName string, helmrequires *Requires) error {
+	content, err := ioutil.ReadFile(fileName)
+	if err != nil {
+		return fmt.Errorf("unable to read file due to: %v", err)
+	}
+	for _, c := range strings.Split(string(content), YamlSeparator) {
+		if err = yaml.Unmarshal([]byte(c), helmrequires); err != nil {
```

Comment:
Is there a good reason to do this? If you want strict, just use that?

Reply:
This was how I did file reading for the bundle file originally, but if there is a different method I'd be interested.

Reply:
It just reads weird because you're unmarshaling, then unmarshaling the same data, but this time with the strict option. I don't get why you're doing that twice?